To: vim_dev@googlegroups.com
Subject: Patch 7.3.292
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.292
Problem:    Crash when using fold markers and selecting a visual block that
	    includes a folded line and goes to end of line. (Sam Lidder)
Solution:   Check for the column to be MAXCOL. (James Vega)
Files:	    src/screen.c


*** ../vim-7.3.291/src/screen.c	2011-08-10 14:32:33.000000000 +0200
--- src/screen.c	2011-09-02 13:58:18.000000000 +0200
***************
*** 2531,2537 ****
  		/* Visual block mode: highlight the chars part of the block */
  		if (wp->w_old_cursor_fcol + txtcol < (colnr_T)W_WIDTH(wp))
  		{
! 		    if (wp->w_old_cursor_lcol + txtcol < (colnr_T)W_WIDTH(wp))
  			len = wp->w_old_cursor_lcol;
  		    else
  			len = W_WIDTH(wp) - txtcol;
--- 2531,2539 ----
  		/* Visual block mode: highlight the chars part of the block */
  		if (wp->w_old_cursor_fcol + txtcol < (colnr_T)W_WIDTH(wp))
  		{
! 		    if (wp->w_old_cursor_lcol != MAXCOL
! 			     && wp->w_old_cursor_lcol + txtcol
! 						       < (colnr_T)W_WIDTH(wp))
  			len = wp->w_old_cursor_lcol;
  		    else
  			len = W_WIDTH(wp) - txtcol;
*** ../vim-7.3.291/src/version.c	2011-09-02 12:27:20.000000000 +0200
--- src/version.c	2011-09-02 14:00:03.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     292,
  /**/

-- 
Hacker: Someone skilled in computer programming (good guy).
Cracker: A hacker that uses his skills to crack software (bad guy).

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
